lua/autorun/server/sv_limit_explosives.lua

```diff
@@ -0,0 +1,52 @@
+local limits = {
+    m9k_proxy = 4
+}
+
+local function getExplosivesTable(ply, class)
+    if limits[class] == nil then return end
+    
+    ply.placedExplosives = ply.placedExplosives or {}
+    ply.placedExplosives[class] = ply.placedExplosives[class] or {}
+    
+    return ply.placedExplosives[class]
+end
+
+local function onExplosiveCreated( owner, ent )
+    if not IsValid( owner ) then return end
+    if not IsValid( ent ) then return end
+
+    local class = ent:GetClass()
+    local explosives = getExplosivesTable( owner, class )
+    if explosives == nil then return end
+    
+    table.insert( explosives, 1, ent )
+    
+    if #explosives > limits[class] then
```

Comment:
I think `table.Count()` is preferred over `#` now?

```suggestion
    if table.Count( explosives ) >= limits[class] then`
```

But you may want to break this up anyway;
```suggestion
    local hasExceededLimit = table.count( explosives ) >= limits[class]

    if hasExceededLimit then
```

Reply:
im creating the table in a way where # should always work

Reply:
Did you note the `>=` vs `>` here?

Reply:
ye, `>=` would only allow 3 proxy mines to exist at once

Reply:
Maybe I'm misunderstanding the check here, but isn't this checking if `spawnedMines > minesLimit` ?

in which case `>` would only prevent new mines from being spawned when the player has `minesLimit + 1` mines?

What am I missing?

Reply:
ye if `spawned mines > the limit` we delete the oldest mine. It isn't preventing mines from being spawned just deleting the oldest mine in the list. the oldest mine should only be deleted if they are over the limit not at the limit.